connect-examples/v2/node_gift-cards/README.md

```diff
@@ -1,309 +1,359 @@
+- `createGiftCardActivity` load funds on the gift card using the [Create gift card activity API](https://developer.squareup.com/reference/square/gift-card-activities-api/create-gift-card-activity). In the request, the activity type is set appropriately. For example, LOAD to load funds or ACTIVATE to activate the gift card.
+
+See code in [gift-card.js](https://github.com/square/connect-api-examples/blob/8db0b397f9c9245d7e9d78f1924f39f59f2a4de2/connect-examples/v2/node_gift-cards/routes/gift-card.js#L140)
+
+```
+router.post("/:gan/add-funds", checkLoginStatus, checkCardOwner, async (req, res, next) => {
+  try {
+    // The following information will come from the request/session.
+    const customerId = req.session.customerId;
+    const amount = req.body.amount;
+    const paymentSource = req.body.cardId;
+    const giftCardState = req.body.state;
+    const gan = req.params.gan;
+
+    // Get the currency to be used for the order/payment.
+    const currency = req.app.locals.currency;
+
+    // The following code runs the order/payment flow.
+    // Await order call, as payment needs order information.
+    const orderRequest = generateOrderRequest(customerId, amount, currency);
+    const { result: { order } } = await ordersApi.createOrder(orderRequest);
+
+    // Extract useful information from the order.
+    const orderId = order.id;
+    const lineItemUid = order.lineItems[0].uid;
+
+    // We have the order response, we can move on to the payment.
+    const paymentRequest = generatePaymentRequest(customerId, amount, currency, paymentSource, orderId);
+    await paymentsApi.createPayment(paymentRequest);
+
+    // Load or activate the gift card based on its current state.
+    // If the gift card is pending, activate it with the amount given.
+    // Otherwise, if the card is already active, load it with the amount given.
+    const giftCardActivity = giftCardState === "PENDING" ? "ACTIVATE" : "LOAD";
+    const giftCardActivityRequest = generateGiftCardActivityRequest(giftCardActivity, gan, orderId, lineItemUid);
+    await giftCardActivitiesApi.createGiftCardActivity(giftCardActivityRequest);
+
+    // Redirect to GET /gift-card/:gan, which will render the card-detail page, with a success message.
+    res.redirect("/gift-card/" + gan + "/?payment=success");
+  } catch (error) {
+    next(error);
+  }
+});
+```
 
+**Note:** Gift cards have different maximum amount limits based on the currency of your seller account. You will not able to pay successfully if a card has reached its limit. In this app we prevent this error from the frontend.
 
+The following [middleware](https://github.com/square/connect-api-examples/blob/master/connect-examples/v2/node_gift-cards/util/middleware.js) are executed to verify permissions:
 
+- `checkLoginStatus` verifies that the customer is logged in
+- `checkCardOwner` verified that the card being accessed belongs to the logged in customer
```

Comment:
I'd switch the order of the note and the middleware description (put the note at the bottom, after describing the middleware).